src/logging.cc

```diff
@@ -1198,8 +1206,19 @@ void LogMessage::Init(const char* file,
                       void (LogMessage::*send_method)()) {
   allocated_ = NULL;
   if (severity != GLOG_FATAL || !exit_on_dfatal) {
+#ifdef GLOG_THREAD_LOCAL_STORAGE
+    // No need for locking, because the data is thread local.
+    if (thread_data_available) {
+      thread_data_available = false;
+      data_ = new (&thread_msg_data) LogMessageData();
+    } else {
+      allocated_ = new LogMessageData();
```

Comment:
I'd add a test case for this code path. Just

LOG(INFO) << LOG(INFO) << "nested LOG";

should be fine.